lib/hierarchical.h

```diff
@@ -185,81 +161,124 @@ class HierarchicalRadiosity {
         bool done = false;
         while (!done) {
             solve_system();
-
             done = !refine_links();
         }
 
-        // Return leaves
-        std::pair<std::vector<Triangle>, std::vector<Color /*rad*/>> out;
+        // store radiosity in mesh
+        store_radiosity_at_triangles();
+        std::cerr << "Triangulating T-vertices." << std::endl;
+        triangulate_t_vertices();
+        store_radiosity_at_vertices();
+    }
+
+    // TODO: Use mesh directly
+    auto triangles() const {
+        std::vector<Triangle> triangles;
+        for (const auto face : mesh_.faces()) {
+            const auto& vs = triangle_vertices(mesh_, face);
+            // copy triangle with minimal information
+            triangles.emplace_back(
+                std::array<Vec, 3>{convert_to_vec(mesh_.point(vs[0])),
+                                   convert_to_vec(mesh_.point(vs[1])),
```

Comment:
Ich meinte eigentlich, warum du `vs` erst zu einem Point machst dun dann wieder in einen Vector konvertierst.